docs/tutorials/TracingCode.adoc

```diff
@@ -0,0 +1,269 @@
++
+image:EditCommand.png[width=666px, height=313px]
+. Let's continue stepping through until we return to `LogicManager#execute()`.
++
+The sequence diagram below shows the details of the execution path through the Logic component.
+Does the execution path you traced in the code so far matches with the diagram?
++
+.Tracing an `edit` command through the Logic component
+image::LogicSequenceDiagram.png[width=1080px, height=476px]
+
+. Now let's see what happens when we call `command#execute()`!
++
+.EditCommand#execute()
+[source, java]
+----
+@Override
+public CommandResult execute(Model model) throws CommandException {
+    ...
+    Person personToEdit = lastShownList.get(index.getZeroBased());
+    Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+    if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+    }
+    model.setPerson(personToEdit, editedPerson);
+    model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+    return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+}
+----
+. As suspected, `command#execute()` does indeed make changes to `model`.
+. We can a closer look at how storage works by repeatedly stepping into the code until we arrive at
+`JsonAddressBook#saveAddressBook()`.
+. Again, it appears that the heavy lifting is delegated. Let's take a look at ``JsonSerializableAddressBook``'s
+constructor.
++
+.JsonSerializableAddressBook#JsonSerializableAddressBook()
+[source, java]
+----
+/**
+ * Converts a given {@code ReadOnlyAddressBook} into this class for Jackson use.
+ *
+ * @param source future changes to this will not affect the created
+ * {@code JsonSerializableAddressBook}.
+ */
+public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
+    persons.addAll(
+        source.getPersonList()
+              .stream()
+              .map(JsonAdaptedPerson::new)
+              .collect(Collectors.toList()));
+}
+----
+. It appears that a `JsonAdaptedPerson` is created for each `Person` and then added to the
+`JsonSerializableAddressBook`.
+. We can continue to step through until we return to `MainWindow#executeCommand()`.
+. Stepping into ``resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());``, we end up in:
++
+.ResultDisplay#setFeedbackToUser()
+[source, java]
+----
+public void setFeedbackToUser(String feedbackToUser) {
+    requireNonNull(feedbackToUser);
+    resultDisplay.setText(feedbackToUser);
+}
+----
+. Finally, we step through until we reach the end of `MainWindow#executeCommand()`.
+
+[discrete]
+== Conclusion
+
+In this tutorial, we traced a valid edit command from raw user input to the result being displayed to the user. From
+this tutorial, you learned more about the inner workings of AddressBook and how the various components mesh together
+to form one cohesive product.
+
+
+{empty} +
+[lead]
+.What's next?
+====
```

Comment:
This renders as an "example": https://deploy-preview-23--addressbook-level35.netlify.com/tutorials/tracingcode